src/components/SeatMap/SeatMap.test.js

```diff
@@ -0,0 +1,554 @@
+            ],
+          }),
+        ],
+        exits: [[]],
+        bulks: [[]],
+      };
+      JetsSeatMapService.prototype.getSeatMapData = jest.fn().mockResolvedValue(seatMapDataForOneDeck);
+
+      setup({
+        flight: flightDetails(),
+        availability: null,
+        passengers: null,
+        currentDeckIndex: 1,
+        configOverrides: { visibleFuselage: false },
+      });
+
+      await waitFor(() => {
+        expect(screen.queryByTestId('jets-nose')).not.toBeInTheDocument();
+        expect(screen.queryByTestId('jets-tail')).not.toBeInTheDocument();
+      });
+    });
+  });
+
+  describe('when the seat map is called with valid flight data', () => {
+    it('should render seat for a single-deck configuration', async () => {
+      const seatMapDataForOneDeck = {
+        params: paramsData(),
+        content: [
+          finalDeck({
+            rows: [
+              row({
+                seats: [
+                  finalSeat({
+                    number: '11A',
+                  }),
+                ],
+              }),
+            ],
+          }),
+        ],
+        exits: [[]],
+        bulks: [[]],
+      };
+
+      JetsSeatMapService.prototype.getSeatMapData = jest.fn().mockResolvedValue(seatMapDataForOneDeck);
+
+      setup({
+        flight: flightDetails(),
+        availability: null,
+        passengers: null,
+        currentDeckIndex: 0, // lower deck
+      });
+
+      await waitFor(() => {
+        const seatNumber = screen.queryByText(/11A/);
+        expect(seatNumber).toBeInTheDocument();
+        expect(screen.queryByTestId('jets-bulk')).not.toBeInTheDocument();
+        expect(screen.queryByTestId('jets-exit')).not.toBeInTheDocument();
+      });
+    });
+
+    it('should render seat availability for a single-deck configuration', async () => {
+      const seatMapDataForOneDeck = {
+        params: paramsData(),
+        content: [
+          finalDeck({
+            rows: [
+              row({
+                seats: [
+                  finalSeat({
+                    number: '11A',
+                    status: 'available',
+                  }),
+                  finalSeat({
+                    number: '11B',
+                    status: 'unavailable',
+                  }),
+                ],
+              }),
+            ],
+          }),
+        ],
+        exits: [[]],
+        bulks: [[]],
+      };
+
+      JetsSeatMapService.prototype.getSeatMapData = jest.fn().mockResolvedValue(seatMapDataForOneDeck);
+
+      setup({
+        flight: flightDetails(),
+        availability: null,
+        passengers: null,
+        currentDeckIndex: 0, // lower deck
+      });
+
+      await waitFor(() => {
+        expect(screen.queryByText(/11A/)).toBeInTheDocument();
+        expect(screen.queryByText(/11B/)).toBeInTheDocument();
+        const seatWrappers = screen.getAllByTestId('jets-seat');
+        expect(seatWrappers[0]).toHaveClass('jets-seat jets-available');
```

Comment:
How can we know that this is the wrapper for the seat 11A?

Reply:
```suggestion
        expect(seatWrappers[0]).toHaveTextContent(/11A/);
        expect(seatWrappers[0]).toHaveClass('jets-seat jets-available');
        expect(seatWrappers[1]).toHaveTextContent(/11B/);
        expect(seatWrappers[1]).toHaveClass('jets-seat jets-unavailable');
```

Adding a `.toHaveTextContent` call should let us be confident in which wrapper we've captured.

Reply:
There are two seats and it should always render in order in the frontend and be retrieved in order, `11A` and then `11B`. I can add the extra check for the text context for clarity, but if the order wasn't a fact, the test in itself would be flay.

Reply:
I've seen that you added Tony's suggestion to check the text context. Thanks!